src/lib/backend.c

```diff
@@ -5,10 +5,45 @@
 #include "backend_esysdb.h"
 #include "backend_fapi.h"
 
-static bool is_fapi_preview_enabled(void) {
+enum backend {
+    backend_unset = 0,
+    backend_error,
+    backend_esysdb,
+    backend_fapi
+};
+
+static enum backend get_backend(void) {
+
+    /* Using a static to only query env once per runtime. */
+    static enum backend backend = backend_unset;
+    if (backend != backend_unset) {
+        return backend;
+    }
 
     const char *env = getenv("FAPI_PREVIEW");
-    return env && ((!strcmp(env, "yes") || !strcmp(env, "true")));
+    if (env && (!strcasecmp(env, "yes") || !strcasecmp(env, "true"))) {
+#ifdef FAPI_3_0
+        LOGW("Environment variable FAPI_PREVIEW is deprecated. Please use "
+             "TPM2_PKCS11_BACKEND=fapi instead.");
+#endif /* FAPI_3_0 */
+        backend = backend_fapi;
+        return backend;
+    }
+
+    env = getenv("TPM2_PKCS11_BACKEND");
+    if (env && !strcasecmp(env, "fapi")) {
+#ifdef FAPI_3_0
+        backend = backend_fapi;
+#else /* FAPI_3_0 */
+        LOGE("Using tpm2-tss (FAPI) < 3.0 does not fully work with FAPI backend. "
+             "Please use FAPI_PREVIEW=true until you can update to tpm2-tss >= 3.0.");
+        backend = backend_error;
```

Comment:
Just make this die in configure.

Reply:
Well, this is the case where we are guessing the SRK to be 81000001 but in non-preview mode. So I'd like to keep this...